src/pybind/mgr/dashboard/frontend/e2e/cluster/hosts.po.ts

```diff
@@ -1,5 +1,46 @@
+import { by, element } from 'protractor';
 import { PageHelper } from '../page-helper.po';
 
 export class HostsPageHelper extends PageHelper {
   pages = { index: '/#/hosts' };
+
+  // function that checks all services links work for first
+  // host in table (if a host is present)
+  check_services_links() {
+    this.navigateTo();
+    let links_tested = 0;
+    this.getTableCount()
+      .getText()
+      .then((hostcount) => {
+        // Only check for services links if there's at least 1 host present
+        if (hostcount.indexOf('0 total') === -1) {
+          // grab services column for first host
+          const services = element.all(by.css('.datatable-body-cell')).get(1);
+          // check is any services links are present
+          services.getText().then((txt) => {
+            // check that text (links) is present in services box
+            if (txt.length !== 0) {
+              const links = services.all(by.css('a.ng-star-inserted'));
+              links.count().then((num_links) => {
+                for (let i = 0; i < num_links; i++) {
+                  // click link, check it worked by looking for changed breadcrumb,
+                  // navigate back to hosts page, repeat until all links checked
+                  links.get(i).click();
+                  expect(this.getBreadcrumbText()).toEqual('Performance Counters');
+                  this.navigateTo();
+                  expect(this.getBreadcrumbText()).toEqual('Hosts');
+                  links_tested++;
+                }
+              });
+            }
+          });
+        }
+      })
+      .then(() => {
+        // if no links were actually tested, share this information
+        if (links_tested === 0) {
```

Comment:
You should expect that the link count is higher than zero here in order to make sure this test will fail when the test requirements are not met.
```suggestion
        expect(links_tested > 0).toBe(true)
```